components/validation/validation-custom-mixin.js

```diff
@@ -0,0 +1,51 @@
+
+export const ValidationCustomMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			failureText: { type: String, attribute: 'failure-text' },
+			for: { type: String }
+		};
+	}
+
+	constructor() {
+		super();
+		this._target = null;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._updateTarget();
+		const connected = new CustomEvent('d2l-validation-custom-connected', { bubbles: true, composed: true });
+		this.dispatchEvent(connected);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._target = null;
+		const disconnected = new CustomEvent('d2l-validation-custom-disconnected', { bubbles: true, composed: true });
+		this.dispatchEvent(disconnected);
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((_, prop) => {
+			if (prop === 'for') {
+				this._updateTarget();
+			}
+		});
+	}
+
+	get target() {
+		return this._target;
+	}
+
+	async validate() {}
+
+	_updateTarget() {
+		const root = this.getRootNode();
+		this._target = root.getElementById(this.for);
```

Comment:
Wondering if we should throw here if we don't find the target? Otherwise page developers may not realize why things later don't behave properly...

Reply:
I like throwing here (or at the very least warning) if `for` is defined and a target isn't found. I am planning on allowing `for` to omitted which will cause whatever handles `d2l-validation-custom-connected` to be the `target` but that shouldn't cause an issue here if we only throw if `for` is set.